### PR TITLE
Add github release flow

### PR DESCRIPTION
**Pull request by niveathika**
## Purpose
$subject

## Goals
Release module after updating the dependency version

## Approach
1. Checkout `release-${NEXT_VERSION}`
1. `ballerinaLangVersion` & `stdlib.*` properties in the gradle.properties are checked & if they are snapshot versions, remove snapshot versions(Assumption dependencies are released beforehand) 
2. The changes are then committed
3. Call release plugin in Gradle
      * Updates module version without snapshot
      * builds and publishes modules
      * Tags the commit with the version
      * Updates module version to next patch snapshot version
      * Pushes changes to origin
4. Send a curl command to create Github release for the newly released tag
5. Post-release work of bumping dependencies to the next snapshot version
6. Merge release branch with master and push to master

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -23,7 +33,25 @@ jobs:
...
-          NEW_VERSION=$(echo "${GITHUB_REF}" | cut -d "/" -f3)
-          echo "New version: ${NEW_VERSION}"
-          echo "Github username: ${GITHUB_ACTOR}"
-          ./gradlew -PpublishToCentral=true -Pversion=${NEW_VERSION} publish
+          ./gradlew release -Prelease.useAutomaticVersion=true
+      - name: Create release
+        run: |
+          curl --request POST 'https://api.github.com/repos/ballerina-platform/module-ballerina-sql/releases' \
+          --header 'Accept: application/vnd.github.v3+json' \
+          --header 'Authorization: Bearer ${{ secrets.BALLERINA_STDLIB_TOKEN }}' \
```
Comment (BuddhiWathsala) [inline .github/workflows/publish-release.yml:41]: Are we going to use the same token OR a new one here? @daneshk @shafreenAnfar @ThisaruGuruge @ldclakmal

File: build-config/release/post-release-dependency-bump.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      then
+        oldVersion=$(echo $value | cut -d'.' -f 3)
+        newVersion=`expr $oldVersion + 1`
+        if [ ! -z "$newVersion" ]
+          then
+            version=$(echo $(echo $value | cut -d'.' -f 1).$(echo $value | cut -d'.' -f 2).$newVersion-SNAPSHOT)
+            sed -i "s/$property=\(.*\)/$property=$version/g" gradle.properties
+          fi
+    fi
+done < gradle.properties
```
Comment (BuddhiWathsala) [inline build-config/release/post-release-dependency-bump.sh:39]:
```suggestion
done < gradle.properties

```

File: .github/workflows/publish-release.yml [hunk 3]
```diff
@@ -23,7 +33,25 @@ jobs:
           packageUser: ${{ github.actor }}
           packagePAT: ${{ secrets.BALLERINA_STDLIB_TOKEN }}
         run: |
-          NEW_VERSION=$(echo "${GITHUB_REF}" | cut -d "/" -f3)
-          echo "New version: ${NEW_VERSION}"
-          echo "Github username: ${GITHUB_ACTOR}"
-          ./gradlew -PpublishToCentral=true -Pversion=${NEW_VERSION} publish
+          ./gradlew release -Prelease.useAutomaticVersion=true
```
Comment (BuddhiWathsala) [inline .github/workflows/publish-release.yml:36]: @niveathika don't we need `./gradlew publish` here?

**Review by BuddhiWathsala — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-release.yml (hunk 3 above)
Comment (niveathika) [reply .github/workflows/publish-release.yml:36]: During the release only we have to publish dont we? After the version of the module also removed SNAPSHOT?

Comment (BuddhiWathsala) [reply .github/workflows/publish-release.yml:36]: Does, `./gradle release` publish to GitHub packages and central(if need)?. IINM, it doesn't.

Comment (niveathika) [reply .github/workflows/publish-release.yml:36]: Yes, publish is called right after the build is checked, https://github.com/ballerina-platform/module-ballerina-sql/blob/de783004cce57f5d57789a581079349c6fa0a76a/build.gradle#L164

File: build-config/release/post-release-dependency-bump.sh [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# ---------------------------------------------------------------------------
```
Comment (ThisaruGuruge) [inline build-config/release/post-release-dependency-bump.sh:1]: When are we running this? I mean which process calls this script?

Comment (niveathika) [reply build-config/release/post-release-dependency-bump.sh:1]: As of now, we decided to hold this. This was supposed to be called by the workflow itself

**Comment by BuddhiWathsala** [comment]
Closing this PR since it is from a forked repo, and due to that builds are failing. Please refer to this updated PR https://github.com/ballerina-platform/module-ballerina-sql/pull/22.

**Pull request closed without merging**